components/server/src/user/env-var-service.ts

```diff
@@ -0,0 +1,256 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { ProjectDB, UserDB } from "@gitpod/gitpod-db/lib";
+import {
+    CommitContext,
+    EnvVar,
+    ProjectEnvVar,
+    UserEnvVar,
+    UserEnvVarValue,
+    WithEnvvarsContext,
+    WorkspaceContext,
+    WorkspaceType,
+} from "@gitpod/gitpod-protocol";
+import { inject, injectable } from "inversify";
+import { Authorizer } from "../authorization/authorizer";
+import { IAnalyticsWriter } from "@gitpod/gitpod-protocol/lib/analytics";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { v4 as uuidv4 } from "uuid";
+import { Config } from "../config";
+import { GuardedResource, ResourceAccessGuard, ResourceAccessOp } from "../auth/resource-access";
+
+export interface ResolvedEnvVars {
+    // all project env vars, censored included always
+    project: ProjectEnvVar[];
+    // merged workspace env vars
+    workspace: EnvVar[];
+}
+
+@injectable()
+export class EnvVarService {
+    constructor(
+        @inject(Config) private readonly config: Config,
+        @inject(UserDB) private readonly userDB: UserDB,
+        @inject(ProjectDB) private readonly projectDB: ProjectDB,
+        @inject(Authorizer) private readonly auth: Authorizer,
+        @inject(IAnalyticsWriter) private readonly analytics: IAnalyticsWriter,
+    ) {}
+
+    async getAllEnvVars(
```

Comment:
I wonder whether it makes sense to rename, to make naming better, i.t. it is not all, but getAllUserEnvVars cc @svenefftinge

Reply:
Yes, definitely. We should align in naming

Reply:
Let's align on

 - listUserEnvVars
 - addUserEnvVar
 - deleteUserEnvVar
 
 same for project, i.e. 
  - listProjectEnvVars
  - ...